subprojects/docs/src/docs/userguide/what_is_gradle.adoc

```diff
@@ -12,133 +12,114 @@
+== Optimizations
 
- * <<implementing_gradle_plugins.adoc#modeling_dsl_like_apis,A custom model>>.
-+
-Gradle allows you to introduce new concepts into a build beyond tasks, files and dependency configurations. You can see this with most language plugins, which add the concept of <<building_java_projects#sec:java_source_sets,_source sets_>> to a build. Appropriate modeling of a build process can greatly improve a build's ease of use and its efficiency.
+Over the years, Gradle has engineered a number of ways to make your builds faster.
+Gradle enables some of these, like the Daemon, by default. You must explicitly enable some other optimizations
+through configuration or build logic. This section highlights some of the most important
+optimizations to know about when you start using Gradle.
 
+=== Configuration Cache
 
-=== 5. Build scripts operate against an API
+The configuration phase is a large up-front cost for all Gradle builds. Every time you build your project, Gradle
+configures the project based on available inputs like build scripts, settings, and system properties. The
+configuration cache allows Gradle to cache the result of this configuration. This can save significant time
+in every build. However, the configuration cache is only compatible with certain syntax.
 
-It's easy to view Gradle's build scripts as executable code, because that's what they are. But that's an implementation detail: well-designed build scripts describe _what_ steps are needed to build the software, not _how_ those steps should do the work. That's a job for custom task types and plugins.
+=== Build Cache
 
-[NOTE]
-====
-There is a common misconception that Gradle's power and flexibility come from the fact that its build scripts are code. This couldn't be further from the truth. It's the underlying model and API that provide the power. As we recommend in our best practices, you should <<authoring_maintainable_build_scripts#sec:avoid_imperative_logic_in_scripts,avoid putting much, if any, imperative logic in your build scripts>>.
-====
+Every time you run a build, Gradle runs a set of tasks.
+The build cache stores the inputs provided to and outputs generated by those tasks.
+When you run Gradle with the build cache enabled, Gradle checks the inputs for each
+task. If the build cache already contains output for a previously completed task's inputs, Gradle reuses that output immediately, without re-running the task.
+Gradle can speed up builds significantly by pulling data from the cache instead of running the same task repeatedly.
+However, the build cache requires annotated inputs and outputs on tasks.
```

Comment:
```suggestion
When you run Gradle with the build cache enabled, Gradle calculates a cache key for the inputs for each
task. If the build cache already contains the outputs for that key, Gradle reuses those outputs immediately without re-running the task.
Enabling the build cache can speed up builds significantly by pulling data from the cache instead of running the same task repeatedly.
```